asyncio_run_in_process/_child.py

```diff
@@ -0,0 +1,163 @@
+import argparse
+import asyncio
+import logging
+import os
+import signal
+import sys
+from typing import (
+    Any,
+    Awaitable,
+    BinaryIO,
+    Callable,
+    Sequence,
+)
+
+from ._utils import (
+    pickle_value,
+    receive_pickled_value,
+)
+from .state import (
+    State,
+)
+from .typing import (
+    TReturn,
+)
+
+logger = logging.getLogger("asyncio-run-in-process")
+
+
+#
+# CLI invocation for subprocesses
+#
+parser = argparse.ArgumentParser(description="asyncio-run-in-process")
+parser.add_argument(
+    "--parent-pid", type=int, required=True, help="The PID of the parent process"
+)
+parser.add_argument(
+    "--fd-read",
+    type=int,
+    required=True,
+    help=(
+        "The file descriptor that the child process can use to read data that "
+        "has been written by the parent process"
+    ),
+)
+parser.add_argument(
+    "--fd-write",
+    type=int,
+    required=True,
+    help=(
+        "The file descriptor that the child process can use for writing data "
+        "meant to be read by the parent process"
+    ),
+)
+
+
+def update_state(to_parent: BinaryIO, state: State) -> None:
+    to_parent.write(state.value.to_bytes(1, 'big'))
+    to_parent.flush()
+
+
+def update_state_initialized(to_parent: BinaryIO) -> None:
+    payload = State.INITIALIZED.value.to_bytes(1, 'big') + os.getpid().to_bytes(4, 'big')
```

Comment:
I checked, [`PID_MAX` is hardcoded to 99998 in OSX](https://apple.stackexchange.com/questions/51119/whats-the-maximum-pid-for-mac-os-x), which is around 2.076 bytes. [On Linux, `PID_MAX_LIMIT`](https://elixir.bootlin.com/linux/latest/source/include/linux/threads.h#L34) is [around 4 million](https://unix.stackexchange.com/questions/16883/what-is-the-maximum-value-of-the-process-id) (Or, 2.75 bytes). Since neither of these are configurable I'm pretty confident they won't change any time soon.

Reply:
Yeah, in theory I could have used 3 bytes but 4 seemed like a nice round number (and originally this was using `struct.pack(...)` which only supports 2/4 byte sizes.

Reply:
Oh, yeah, definitely not suggesting it should be 3! Chiming in because I also didn't know how many bytes was reasonable, but it looks like 4 is more than enough.